MoltenVK/MoltenVK/GPUObjects/MVKPixelFormats.mm

```diff
@@ -484,7 +485,13 @@
 												VK_IMAGE_USAGE_TRANSFER_DST_BIT)) &&	// Scaling a BLIT may use rendering.
 		mvkIsAnyFlagEnabled(mtlFmtCaps, (kMVKMTLFmtCapsColorAtt | kMVKMTLFmtCapsDSAtt))) {
 
-		mvkEnableFlags(mtlUsage, MTLTextureUsageRenderTarget);
+#if MVK_MACOS
+        if(!isLinear) {
+            mvkEnableFlags(mtlUsage, MTLTextureUsageRenderTarget);
+        }
+#else
+        mvkEnableFlags(mtlUsage, MTLTextureUsageRenderTarget);
+#endif
```

Comment:
This is an interesting solution. And there are examples of us handling other OS variations along these lines.

I am a bit concerned that silently handling this in `MVKPixelFormats::getMTLTextureUsage()` introduces a functional change that might surprise the app with unexpected behaviour. It might just defer the problem, as any attempt by the app to use the linear texture as a BLIT destination will cause a Metal error.

In `MVKImage::validateLinear()`, we perform the equivalent test to see if the linear image might be used as an attachment, and we report an error if so. 

I'm wondering if adding a similar check and error report to `MVKImage::validateLinear()` if `VK_IMAGE_USAGE_TRANSFER_DST_BIT` is set would be an appropriate answer.

The spec says one of the restrictions for linear images is:

>usage only includes `VK_IMAGE_USAGE_TRANSFER_SRC_BIT` and/or `VK_IMAGE_USAGE_TRANSFER_DST_BIT`

And I recognize that `VK_IMAGE_USAGE_TRANSFER_DST_BIT` covers both BLIT and copy...so perhaps leaving the image usage at only `VK_IMAGE_USAGE_TRANSFER_SRC_BIT` usage might be too restrictive.

Thoughts?

Reply:
If silently eliminating `MTLTextureUsageRenderTarget` as is done here does prove useful and necessary, we should add a validation error test to `MVKCmdBlitImage<N>::setContent()` similar to what is done for a depth/stencil format for the same reasons (but only for macOS).

I can add that if needed.

Reply:
Adding a check to `MVKImage::validateLinear()` for `VK_IMAGE_USAGE_TRANSFER_DST_BIT` would just move the error to a different place for me. 

My issue is that I have a HOST_VISIBLE | HOST_COHERENT 2D texture that I map/unmap in order to write from the CPU, and it then gets SHADER_READ by a shader. This texture gets assigned a render target usage in MVK and triggers the Metal validation error.

I'm not sure what is the correct solution for MVK's design, I just know the above is causing problems for my use case.

Reply:
>HOST_VISIBLE | HOST_COHERENT 2D texture that I map/unmap in order to write from the CPU, and it then gets SHADER_READ by a shader

Yes...that's a typical use of linear textures. With that requirement, why does the image need `VK_IMAGE_USAGE_TRANSFER_DST_BIT`, which would imply you intend to execute a [transfer command](https://www.khronos.org/registry/vulkan/specs/1.0-extensions/html/vkspec.html#synchronization-pipeline-stages-transfer) to it (see definition of `VK_PIPELINE_STAGE_TRANSFER_BIT`)?

Sorry for belabouring this...and I'm not questioning your use of `VK_IMAGE_USAGE_TRANSFER_DST_BIT` per se, because as I mention above, Vulkan indeed allows both `VK_IMAGE_USAGE_TRANSFER_SRC_BIT` and `VK_IMAGE_USAGE_TRANSFER_DST_BIT` for linear textures.

But in this case, the use case matters...because I'm trying to determine what the appropriate use cases should be for a linear texture, given the restrictions of Metal's transfer command capabilities, and whether it would be a fair restriction to disallow `VK_IMAGE_USAGE_TRANSFER_DST_BIT` for linear textures on macOS.

@cdavis5e I'm happy to hear your thoughts on these options too.

Reply:
We don't actually use the `VK_IMAGE_USAGE_TRANSFER_DST_BIT`. But our user-facing API is higher level and we assign the flag for the eventuality the user decides to execute the transfer operation.

Reply:
Okay. Accepting this, and then I'll add the [suggested validation code](https://github.com/KhronosGroup/MoltenVK/pull/972#discussion_r459593524).